pkg/oc/cli/tag/tag.go

```diff
@@ -417,10 +403,18 @@ func (o TagOptions) Run() error {
 				}
 
 				// The user wants to delete a spec tag.
-				if _, ok := target.Spec.Tags[destTag]; !ok {
+				if _, ok := imageutil.SpecHasTag(target, destTag); !ok {
 					return fmt.Errorf("destination tag %s/%s does not exist.\n", o.destNamespace[i], destNameAndTag)
 				}
-				delete(target.Spec.Tags, destTag)
+				// delete tag
+				tags := []imagev1.TagReference{}
+				for i := range target.Spec.Tags {
+					t := target.Spec.Tags[i]
+					if t.Name != destTag {
```

Comment:
nit: might improve readability a bit if we `continue` on the inverse condition:

if t.Name == destTag {
  continue
}

tags = append(tags, t)

Reply:
I find the current approach more readable, so I'll leave it as is.